### ISSUE TITLE
False positive E302 if header of the top function spans more than two lines

### ISSUE THREAD

**Issue by east825** [PyCQA/pycodestyle#473]
If parameters of the first function defined in a file (no comments or other statements precede it) span more than two lines (with or without hanging indent) pep8.py falsely reports "E302 expected 2 blank lines, found 0" . However if the same function is reformatted so that its parameters are in row or occupy only two lines, there is no such warning.

For instance, no warning is given in the following cases

``` python
def func(x, y, z):
    pass
```

``` python
def func(
        x, y, z):
    pass
```

``` python
def func(x, 
         y, z):
    pass
```

Nonetheless, pep8.py complains about the first line in these cases:

``` python
def func(x,  # E302 expected 2 blank lines, found 0 
         y,
         z):
    pass
```

``` python
def func(  # E302 expected 2 blank lines, found 0
        x,
        y, z):
    pass
```

**Comment by sigmavirus24** [comment]
- What version of pep8 are you using?
- What version of python are you running pep8 with?

**Comment by east825** [comment]
I've checked with CPython 2.7.7 and pep8 1.7.0.

**Comment by IanLee1521** [comment]
I can confirm this issue with commit 91717019e4a98bf9bb47cd2819035ff14398e773 and the failing tests [in Travis](https://travis-ci.org/PyCQA/pep8/builds/109777028).

**Comment by asottile** [comment]
this seems ~related to #412 -- might be able to fix both at the same time

**Comment by asottile** [comment]
I've created #919 to attempt to resolve this, please let me know if this fixes it (and/or review the patch)
